src/main/java/com/puppycrawl/tools/checkstyle/DefaultConfiguration.java

```diff
@@ -103,8 +102,8 @@ public String getProperty(String propertyName) throws CheckstyleException {
 
     @Override
     public Configuration[] getChildren() {
-        return children.toArray(
-                EMPTY_CONFIGURATION_ARRAY);
+        final Configuration @ArrayLen(0) [] emptyArray = new Configuration[0];
+        return children.toArray(emptyArray);
```

Comment:
```suggestion
        return children.toArray(new Configuration[0]);
```
The type looks quite impressive, do we need the explizit boilerplate typing here for specific reasons?

This will be done anyway in #16488 as violating against RSPEC-S1488

introduce InlineVariableCheck - Local variables should not be declared and then immediately returned or thrown - (RSPEC-S1488)